src/verify.cc

```diff
@@ -68,6 +68,39 @@ bool verifySignatureRSA(RSA* key, const EVP_MD* md, absl::string_view signature,
                             castToUChar(signed_data), signed_data.length());
 }
 
+bool verifySignatureRSAPSS(RSA* key, const EVP_MD* md, const uint8_t* signature,
+                           size_t signature_len, const uint8_t* signed_data,
+                           size_t signed_data_len) {
+  if (key == nullptr || md == nullptr || signature == nullptr ||
+      signed_data == nullptr) {
+    return false;
+  }
+  bssl::UniquePtr<EVP_PKEY> evp_pkey(EVP_PKEY_new());
+  if (EVP_PKEY_set1_RSA(evp_pkey.get(), key) != 1) {
+    return false;
+  }
+
+  bssl::UniquePtr<EVP_MD_CTX> md_ctx(EVP_MD_CTX_create());
+  EVP_PKEY_CTX *pctx;
```

Comment:
do we need to free pctx?

Reply:
I believe we do not have to free `pctx`. I was referring to [this part of boringssl code](https://github.com/google/boringssl/blob/eda849d2e6e6a15a5a4dc728568ec12f21ebfb6d/crypto/x509/rsa_pss.c#L282). Although that's in C, the `setup_ctx` function doesn't free `pctx`, and doesn't return `pctx` either, so it couldn't have been freed elsewhere.

Over here in [BoringSSL docs](https://github.com/google/boringssl/blob/eda849d2e6e6a15a5a4dc728568ec12f21ebfb6d/include/openssl/evp.h#L335) it says that the `EVP_PKEY_CTX` of the signing operation will be written to `*pctx`, and in the implementation [here](https://github.com/google/boringssl/blob/eda849d2e6e6a15a5a4dc728568ec12f21ebfb6d/crypto/evp/digestsign.c#L79), we see that the `ctx->pctx` is assigned to `pctx`. 

`EVP_MD_CTX_cleanup` will free `ctx->pctx` ([here](https://github.com/google/boringssl/blob/eda849d2e6e6a15a5a4dc728568ec12f21ebfb6d/crypto/fipsmodule/digest/digest.c#L94:5)) and I'm assuming the use of `UniquePtr` handles calling of `EVP_MD_CTX_cleanup`? Let me look into that a bit more.

Reply:
Yup okay there's a 

```
BORINGSSL_MAKE_DELETER(EVP_MD_CTX, EVP_MD_CTX_free)
```

[here](https://github.com/google/boringssl/blob/eda849d2e6e6a15a5a4dc728568ec12f21ebfb6d/include/openssl/evp.h#L1074) which makes `EVP_MD_CTX_free` the deleter for `EVP_MD_CTX`, and so freeing of `pctx` is handled.

We don't have to specially free `pctx`, it's like an additional reference :)

Reply:
OK,  then please add a comment on top of pctx, something like following:
// pctx is owned by md_ctx, not need to free it separately.